XenAdmin/Controls/Ballooning/VMMemoryControlsBasic.cs

```diff
@@ -100,21 +100,15 @@ protected override void OnPaint(PaintEventArgs e)
                     }
                     if (identical)
                     {
-                        switch (vm0.GetVirtualisationStatus)
-                        {
-                            case VM.VirtualisationStatus.OPTIMIZED:
-                                labelDMCUnavailable.Text = Messages.DMC_UNAVAILABLE_NOTSUPPORTED_PLURAL;
-                                break;
-                            case VM.VirtualisationStatus.PV_DRIVERS_NOT_INSTALLED:
-                                labelDMCUnavailable.Text = Messages.DMC_UNAVAILABLE_NOTOOLS_PLURAL;
-                                break;
-                            case VM.VirtualisationStatus.PV_DRIVERS_OUT_OF_DATE:
-                                labelDMCUnavailable.Text = Messages.DMC_UNAVAILABLE_OLDTOOLS_PLURAL;
-                                break;
-                            default:  // VM.VirtualisationStatus.UNKNOWN
-                                labelDMCUnavailable.Text = Messages.DMC_UNAVAILABLE_VMS;
-                                break;
-                        }
+                        var status = vm0.GetVirtualisationStatus;
+                        if (status.HasFlag(VM.VirtualisationStatus.IO_DRIVERS_INSTALLED) && status.HasFlag(VM.VirtualisationStatus.MANAGEMENT_INSTALLED))

```

Comment:
I think this line should only test for I/O drivers, otherwise you will get the wrong error message in the case you have I/O drivers but no management agent.